pycalphad/core/minimizer.pyx

```diff
@@ -27,6 +28,17 @@ cdef void lstsq(double *A, int M, int N, double* x, double rcond) nogil:
         for i in range(N):
             x[i] = -1e19
 
+cpdef void lstsq_check_infeasible(double[:,::] A, double[::] b, double[::] out_x):
+    # Note: This function will destroy input matrix A
+    A_copy = np.copy(A)
+    b_copy = np.copy(b)
+    lstsq(&A[0,0], A.shape[0], A.shape[1],
+        &out_x[0], 1e-16)
+    residual = np.sum(np.square(np.dot(A_copy, out_x) - b_copy))
+    if residual > 1e-6:
+        # lstsq solution is spurious; throw it away
+        out_x[:] = np.nan
+
```

Comment:
It's not a blocker and I'm expecting any action, but it would be nice to have a solution to avoid a copy. This should be infrequent enough because we don't expect to call the `_differential` functions that use this in tight inner loops.

Reply:
If you don't need to check the residual, you can call `lstsq` to avoid the copy. Otherwise, this copy has to occur somewhere.